src/main/java/net/openhft/chronicle/queue/impl/single/StoreAppender.java

```diff
@@ -427,22 +427,28 @@ private StoreAppender.StoreAppenderContext prepareAndReturnWriteContext(boolean
             prepareDoubleBuffer();
         } else {
             writeLock.lock();
-            int cycle = queue.cycle();
-            if (wire == null)
-                setWireIfNull(cycle);
 
-            if (this.cycle != cycle)
-                rollCycleTo(cycle);
+            try {
+                int cycle = queue.cycle();
+                if (wire == null)
+                    setWireIfNull(cycle);
 
-            long safeLength = queue.overlapSize();
-            resetPosition(false);
-            assert !QueueSystemProperties.CHECK_INDEX || checkWritePositionHeaderNumber();
+                if (this.cycle != cycle)
+                    rollCycleTo(cycle);
+
+                long safeLength = queue.overlapSize();
+                resetPosition(false);
+                assert !QueueSystemProperties.CHECK_INDEX || checkWritePositionHeaderNumber();
 
-            // sets the writeLimit based on the safeLength
-            openContext(metaData, safeLength);
+                // sets the writeLimit based on the safeLength
+                openContext(metaData, safeLength);
 
-            // Move readPosition to the start of the context. i.e. readRemaining() == 0
-            wire.bytes().readPosition(wire.bytes().writePosition());
+                // Move readPosition to the start of the context. i.e. readRemaining() == 0
+                wire.bytes().readPosition(wire.bytes().writePosition());
+            } catch (WriteAfterEOFException e) {
```

Comment:
I wonder if it's safe to generalise this to RuntimeException. It feels a bit less hacky to only unlock for a specific failure we know the consequences of.

Reply:
I had to change it to `RuntimeException` because the Java 11 build failed mysteriously when I caught `WriteAfterEOFException`